src/test/java/com/atm/repository/TransactionRepositoryTest.java

```diff
@@ -31,7 +32,31 @@ public void testFindTransactionWithinRangeDate_returnSuccess() {
             transactionRepository.save(transaction);
         }
 
-        List<Transaction> trxs = transactionRepository.findByTransactionDateBetween("2019-12-01 10:18:29", "2019-12-02 10:18:29");
+        List<Transaction> trxs = transactionRepository.findByTransactionDateBetweenOrderByTransactionDateDesc(LocalDateTime.of(2019,12,01, 10,18,29), LocalDateTime.of(2019,12,02,10,18,29));
         Assert.assertEquals(2, trxs.size());
     }
+
+    @Test
+    public void testFindTop10Transaction_returnSuccess() {
+        List<Transaction> transactions = new ArrayList<>();
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,02,10,18,29), "Withdraw", "112233", null, Double.valueOf(10), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,01,10,18,29), "Withdraw", "112233", null, Double.valueOf(20), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,11,30,10,18,29), "Withdraw", "112233", null, Double.valueOf(30), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,02,10,18,29), "Withdraw", "112233", null, Double.valueOf(10), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,01,10,18,29), "Withdraw", "112233", null, Double.valueOf(20), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,11,30,10,18,29), "Withdraw", "112233", null, Double.valueOf(30), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,05,10,18,29), "Withdraw", "112233", null, Double.valueOf(10), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,01,10,18,29), "Withdraw", "112233", null, Double.valueOf(20), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,11,30,10,18,29), "Withdraw", "112233", null, Double.valueOf(30), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,02,10,18,29), "Withdraw", "112244", "112233", Double.valueOf(10), "019182"));
+        transactions.add(new Transaction(LocalDateTime.of(2019,12,01,10,18,29), "Withdraw", "112233", null, Double.valueOf(20), null));
+        transactions.add(new Transaction(LocalDateTime.of(2019,11,30,10,18,29), "Withdraw", "112244", "112233", Double.valueOf(30), "223394"));
+        for (Transaction transaction : transactions) {
+            transactionRepository.save(transaction);
+        }
```

Comment:
This can be improved by using instead `transactionRepository.save(transactions)` or `transactionRepository.saveAll(transactions)` (it will depend on your JPA version). The problem is, your current interface extends from CrudRepository. Take a look at this article for some clarification https://www.baeldung.com/spring-data-repositories